service/migration_manager.cc

```diff
@@ -865,7 +865,7 @@ future<std::vector<mutation>> prepare_view_update_announcement(storage_proxy& sp
 future<std::vector<mutation>> prepare_view_drop_announcement(storage_proxy& sp, const sstring& ks_name, const sstring& cf_name, api::timestamp_type ts) {
     auto& db = sp.local_db();
     try {
-        auto& view = db.find_column_family(ks_name, cf_name).schema();
+        const auto& view = db.find_column_family(ks_name, cf_name).schema();
         if (!view->is_view()) {
```

Comment:
This check is done each time we construct a `view_ptr` from a view.
It makes sense to check it first but another form is also possible:
- wrap the code with `try catch` and re-throw with the "cannot use drop... on table" message.
- use only view_ptr and don't use view at all.
This is not a must, just an idea.

Reply:
> This check is done each time we construct a `view_ptr` from a view.
>  It makes sense to check it first but another form is also possible:
> 
>     * wrap the code with `try catch` and re-throw with the "cannot use drop... on table" message.
> 

i see your concern comes from the duplication of the check in the constructor of `view_ptr` and `prepare_view_drop_announcement()`. but they serve for different purpose at different levels of abstraction. the ways how they handle the `!is_view()` are different as well. i don't think we should trade one for another.

>     * use only view_ptr and don't use view at all.
>       This is not a must, just an idea.

then we have to introduce another variant of `make_drop_view_mutations()`. if we see more potential callers of this variant, we can add it. but i cannot find another one at this moment.

also, i'd prefer limiting the scope of this pull request.